packages/api/src/routes/internal/medical/patient.ts

```diff
@@ -1170,4 +1171,33 @@ router.post(
   })
 );
 
+/** ---------------------------------------------------------------------------
+ * DELETE /internal/patient/:id/external-data
+ *
+ * Resets the external data for a specific patient and HIE source by removing it completely.
```

Comment:
nit: "the external data" is a bit generic, for a sec I was questioning whether it would nuke all the patient data in S3

can we be a bit more explicit as to what this means? ie the metadata on the patient table corresponding to the specified source